src/metabase/models/table.clj

```diff
@@ -296,3 +297,117 @@
 (defmethod audit-log/model-details Table
   [table _event-type]
   (select-keys table [:id :name :db_id]))
+
+;;;; ------------------------------------------------- Search ----------------------------------------------------------
+
+(search/define-spec "table"
```

Comment:
it's hard to wrap my head around why we need all these keys, it'll be especially hard for other teams rather than our defining a spec.

do you think we can strip it to the bare minimum? if not I'd make these key more explicit so it aids readability

Reply:
Unfortunately we just do need all this data to populate the index.

Beyond describing each one in the docstring, we could also some context in the Malli types as well, would this satisfy you?

We could move `:archived`, `:created-at` and `:updated-at` into `:context`, does that kind of change spark joy?

We could also make `:render-terms` a typed map as well actually, that could help make things a bit less daunting and more discoverable. I've already made things more svelte by implicitly including keys mentioned elsewhere in the actual render terms 😬 

The only required keys will be `:model`, `:context`, and `:search-terms`. I've already made things a bit